lib/Elastica/Scroll.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Elastica;
+
+/**
+ * Scroll Iterator
+ *
+ * @author Manuel Andreo Garcia <andreo.garcia@gmail.com>
+ *
+ * @link http://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
+ */
+class Scroll implements \Iterator
+{
+    /**
+     * @var string
+     */
+    public $expiryTime;
+
+    /**
+     * @var Search
+     */
+    protected $_search;
+
+    /**
+     * @var null|string
+     */
+    protected $_nextScrollId = null;
+
+    /**
+     * @var null|ResultSet
+     */
+    protected $_currentResultSet = null;
+
+    /**
+     * 0: scroll<br>
+     * 1: scroll id<br>
+     * 2: search type.
+     *
+     * @var array
+     */
+    protected $_options = array(null, null, null);
```

Comment:
@hasumedic Currently the convention is arrays. This is mainly related to the history of Elastica as it started very simple and grew together with elasticsearch. If we change to ArrayObjects, we should consider a change across the full library. @ewgRa brought up this issue and I just realised I never responded to it :-( https://github.com/ruflin/Elastica/issues/783#issuecomment-75366015 I suggest to continue this discussion in #783 an until then we use arrays.